plugins/woocommerce/includes/tracks/class-wc-tracks.php

```diff
@@ -115,6 +115,27 @@ public static function record_event( $event_name, $event_properties = array() )
 		return $event_obj->record();
 	}
 
+	/**
+	 * Track when the user attempts to toggle
+	 * woocommerce_allow_tracking option.
+	 *
+	 * @param string $prev_value The previous value for the setting. 'yes' or 'no'.
+	 * @param string $new_value The new value for the setting. 'yes' or 'no'.
+	 * @param string $context Which avenue the user utilized to toggle.
+	 */
+	public static function track_woocommerce_allow_tracking_toggled( $prev_value, $new_value, $context = 'settings' ) {
```

Comment:
I don't see any method in the class or other similar classes supporting strict types. So that's why I decided to keep it out scope of this work

Reply:
Yup, WC has a lot of code with loose types / type checking which mainly stems from needing to support older PHP versions or it just being older code. I just think it's nice to be stricter when we can and it doesn't have any negative side effects. Otherwise we'll never end up modernizing any code, but like you mentioned it's not necessary.